### PR TITLE
feat: add page convenience methods for textContent and getAttribute

### LINKED ISSUE #2143: [Feature] expose textContent, innerText, getAttribute on page/frame
Missing these convenience methods.

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch adds:
- `page.innerText()` / `frame.innerText()`
- `page.innerHTML()` / `frame.innerHTML()`
- `page.textContent()` / `frame.textContent()`
- `page.getAttribute()` / `frame.getAttribute()`

Fixes #2143

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1275,6 +1279,15 @@ Returns frame matching the specified criteria. Either `name` or `url` must be sp
 #### page.frames()
 - returns: <[Array]<[Frame]>> An array of all frames attached to the page.
 
+#### page.getAttribute(selector, name[, options])
+- `selector` <[string]> A selector to search for an element. If there are multiple elements satisfying the selector, the first will be checked. See [working with selectors](#working-with-selectors) for more details.
```
Comment (dgozman) [inline docs/api.md:1283]: "will be checked" is probably a copy-paste from somewhere 😄 Same in other methods.

File: docs/api.md [hunk 2]
```diff
@@ -1345,6 +1358,22 @@ If there's no element matching `selector`, the method waits until a matching ele
 
 Shortcut for [page.mainFrame().hover(selector[, options])](#framehoverselector-options).
 
+#### page.innerHTML(selector[, options])
+- `selector` <[string]> A selector to search for an element. If there are multiple elements satisfying the selector, the first will be checked. See [working with selectors](#working-with-selectors) for more details.
+- `options` <[Object]>
+  - `timeout` <[number]> Maximum time in milliseconds, defaults to 30 seconds, pass `0` to disable timeout. The default value can be changed by using the [browserContext.setDefaultTimeout(timeout)](#browsercontextsetdefaulttimeouttimeout) or [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) methods.
+- returns: <[Promise]<null|[string]>>
+
+Resolves to the `node.innerHTML`.
```
Comment (dgozman) [inline docs/api.md:1367]: We always resolve to elements, make this `element.innerHTML`? Same in other methods.

File: src/page.ts [hunk 3]
```diff
@@ -452,6 +452,22 @@ export class Page extends ExtendedEventEmitter implements InnerLogger {
     return this.mainFrame().focus(selector, options);
   }
 
+  async textContent(selector: string, options?: types.TimeoutOptions): Promise<string | null> {
+    return this.mainFrame().textContent(selector, options);
+  }
+
+  async innerText(selector: string, options?: types.TimeoutOptions): Promise<string | null> {
```
Comment (dgozman) [inline src/page.ts:459]: How can innerText or innerHTML be null?

Also, for textContent, would it be much more convenient to convert null into empty string?

Comment (aslushnikov) [reply src/page.ts:459]: these all are actually always strings; casted.

File: src/frames.ts [hunk 4]
```diff
@@ -730,6 +730,26 @@ export class Frame {
...
   }
 
+  async textContent(selector: string, options: types.TimeoutOptions = {}): Promise<string> {
+    return await this._retryWithSelectorIfNotConnected('textContent', selector, options,
+        (handle, deadline) => handle.textContent() as Promise<string>);
+  }
+
+  async innerText(selector: string, options: types.TimeoutOptions = {}): Promise<string> {
+    return await this._retryWithSelectorIfNotConnected('innerText', selector, options,
+        (handle, deadline) => handle.innerText() as Promise<string>);
```
Comment (dgozman) [inline src/frames.ts:740]: Why don't we update ElementHandle.innerText() and ElementHandle.innerHTML() to return string as well?

File: src/frames.ts [hunk 5]
```diff
@@ -730,6 +730,26 @@ export class Frame {
         (handle, deadline) => handle.focus());
   }
 
+  async textContent(selector: string, options: types.TimeoutOptions = {}): Promise<string> {
+    return await this._retryWithSelectorIfNotConnected('textContent', selector, options,
+        (handle, deadline) => handle.textContent() as Promise<string>);
```
Comment (dgozman) [inline src/frames.ts:735]: Hmm... We should have the same return type here and in ElementHandle.textContent.

**Review by dgozman — APPROVED** [review]

File: src/frames.ts (hunk 4 above)
Comment (aslushnikov) [reply src/frames.ts:740]: done

File: src/frames.ts (hunk 5 above)
Comment (aslushnikov) [reply src/frames.ts:735]: Done.

**Review by dgozman — APPROVED** [review]

**Pull request merged**